### PR TITLE
feat: Update release.yml

### PR DESCRIPTION
**Pull request by AgrimVerma**
no issue link in description

### CODE REVIEW

**Comment by AgrimVerma** [comment]
Review:

Clarity of the description:
The description in this pull request is not very clear. It simply states that there is no issue link in the description, followed by a link to an issue (AG-4). It is not clear what the issue is or what changes are being made in this pull request. The description should provide a brief overview of the changes and the purpose of the pull request.

Code quality considerations:
It is difficult to assess the code quality without any context or code changes provided in the description. However, it is important to ensure that the code changes made in this pull request follow best practices and adhere to coding standards. It is also important to consider the impact of these changes on the overall codebase and make sure they do not introduce any potential issues.

Potential issues with the code:
Without any code changes or context provided, it is not possible to identify any potential issues with the code. However, it is important to thoroughly review the code changes and ensure that they do not introduce any bugs or cause any conflicts with existing code.

Suggested improvements:
1. Provide a clear and concise description of the changes in this pull request.
2. Include the code changes or provide a link to the specific files being modified.
3. Follow best practices and coding

**Comment by AgrimVerma** [comment]
Review:

- Clarity of the description: The description is clear and concise, stating that there is no issue link in the description and providing a link to the relevant issue (AG-4) on the issue tracker. It is easy to understand and sets clear expectations for the reviewer.

- Code quality considerations: There is no code included in the pull request description, so it is difficult to assess the code quality. However, the fact that the issue is related to a specific issue on the issue tracker (AG-4) suggests that the code changes are targeted and not just random changes.

- Potential issues with the code: As mentioned earlier, there is no code included in the description, so it is difficult to identify potential issues with the code. However, if the code changes are related to a specific issue, it would be helpful to include a brief summary of the changes or a link to the specific code changes in the description to provide context for the reviewer.

- Suggested improvements: One improvement that could be suggested is to include a brief summary of the code changes or a link to the specific code changes in the description. This would provide more context for the reviewer and help them understand the changes being made. Additionally, it would be helpful to include any relevant test cases

**Comment by AgrimVerma** [comment]
As a code reviewer, it's challenging to provide a comprehensive and detailed review based on the provided pull request description. The description is very minimal and lacks the necessary information to effectively evaluate the code changes. Here are some general observations:

1. Clarity of the description: The description is not clear at all. It only mentions a Jira issue link but does not provide any context about what changes have been made in the code, why they were made, or how they affect the overall project. A good pull request description should provide a clear, concise summary of the changes and the reason behind them.

2. Code quality considerations: Without seeing the actual code or knowing what changes have been made, it's impossible to comment on the code quality. The pull request should include details about what was changed in the code, why it was changed, and how it was tested.

3. Potential security or performance issues: Similarly, without the actual code or a clear description of the changes, it's impossible to identify any potential security or performance issues.

4. Adherence to best practices: The lack of a clear, detailed description in the pull request does not adhere to best practices for code reviews. A good pull request should include a clear, concise description of the changes, the reason behind them, and how they were tested.

5. Suggested improvements or refactoring: Without seeing the actual code or knowing what changes have been made, it's impossible to suggest any improvements or refactoring.

6. Any concerns related to documentation or comments: The lack of a clear, detailed description in the pull request raises concerns about the overall quality of the documentation and comments in the code. Good documentation and comments are essential for understanding the code and making effective code reviews.

In conclusion, the pull request description needs significant improvements. It should provide a clear, concise summary of the changes, the reason behind them, and how they were tested. This will help the reviewer understand the changes and make an effective review.

**Comment by AgrimVerma** [comment]
1. **Clarity of the description**: The description of the pull request is not very clear. It only mentions an issue link, but does not provide any context or explanation of what the changes are about. It would be better to include a brief summary of the changes made, the reason for the changes, and the expected impact of the changes.

2. **Code quality considerations**: The code seems to be well-structured and follows a logical flow. However, there are a few areas where readability could be improved. For instance, the use of complex one-liners, especially with the `awk` and `sed` commands, makes the code harder to understand. Breaking these down into multiple lines or adding comments to explain what they do would improve readability.

3. **Potential security or performance issues**: The code makes use of environment variables for sensitive information, which is a good practice. However, it's important to ensure that these environment variables are properly secured, especially since they seem to include API tokens. There don't appear to be any major performance issues, but the use of `grep`, `sed`, and `awk` in a pipeline could potentially be optimized.

4. **Adherence to best practices**: The code generally adheres to best practices, such as the use of environment variables for sensitive data. However, the lack of comments explaining what the code does is a deviation from best practices. 

5. **Suggested improvements or refactoring**: Consider adding more comments to the code to explain what each section does, especially for complex one-liners. Also, consider breaking down complex one-liners into multiple lines for better readability. The error handling could also be improved. Currently, if no JIRA issue key is found, the code simply sets `jira_issue_key` to 'none' and continues execution. It would be better to handle this error more explicitly.

6. **Any concerns related to documentation or comments**: As mentioned above, the code lacks sufficient comments. Adding comments would make the code easier to understand and maintain. Also, the pull request description should be updated to provide more context and explanation about the changes.

**Comment by AgrimVerma** [comment]
1. **Clarity of the description**: The description of the pull request is still not clear. It only mentions an issue link, but does not provide any context or explanation of what the changes are about. It would be better to include a brief summary of the changes made, the reason for the changes, and the expected impact of the changes.

2. **Code quality considerations**: The code seems to be well-structured and follows a logical flow. However, the use of complex one-liners, especially with the `awk` and `sed` commands, makes the code harder to understand. Breaking these down into multiple lines or adding comments to explain what they do would improve readability. The changes in the pull request seem to handle the case when no JIRA issue key is found in a more explicit way, which is a good improvement.

3. **Potential security or performance issues**: The code makes use of environment variables for sensitive information, which is a good practice. However, it's important to ensure that these environment variables are properly secured, especially since they seem to include API tokens. There don't appear to be any major performance issues, but the use of `grep`, `sed`, and `awk` in a pipeline could potentially be optimized.

4. **Adherence to best practices**: The code generally adheres to best practices, such as the use of environment variables for sensitive data. However, the lack of comments explaining what the code does is a deviation from best practices. 

5. **Suggested improvements or refactoring**: Consider adding more comments to the code to explain what each section does, especially for complex one-liners. Also, consider breaking down complex one-liners into multiple lines for better readability. The error handling could also be improved. Currently, if no JIRA issue key is found, the code simply sets `jira_issue_key` to 'none' and continues execution. It would be better to handle this error more explicitly.

6. **Any concerns related to documentation or comments**: As mentioned above, the code lacks sufficient comments. Adding comments would make the code easier to understand and maintain. Also, the pull request description should be updated to provide more context and explanation about the changes.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -145,13 +145,15 @@ jobs:
           jira_issue_key=$(echo "${{ fromJson(steps.get_pr_data.outputs.result).body }}" | grep -oE '[A-Z]+-[0-9]+' | head -n 1)
```
Comment (AgrimVerma) [inline .github/workflows/release.yml:145]: The code changes are generally well-structured and follow good practices. However, there are a few points that can be improved:

1. **Code Quality Considerations:**
   - The code is quite verbose and could benefit from some refactoring. For instance, the creation of the `RELEASE_NOTES` string is duplicated in the `if` and `else` blocks. This could be extracted into a function to avoid repetition.
   - The use of `sed` and `awk` commands can be quite complex and hard to understand. Consider adding comments to explain what each command is doing, especially for complex regular expressions.
   - The use of `echo` to print debug information like `echo "RELEASE_NOTES: $RELEASE_NOTES"` could be replaced with a proper logging mechanism to provide more control over the log levels and output.

2. **Potential Security or Performance Issues:**
   - The code uses `curl` to make a request to a JIRA API with a username and password. This could potentially expose sensitive information if the output is logged or intercepted. Consider using secure methods to store and use sensitive information.
   - The use of `grep`, `sed`, and `awk` commands in a pipeline can be inefficient for large amounts of data. Consider using a single tool like `awk` or a scripting language like Python to process the data.

3. **Adherence to Best Practices:**
   - The code checks if the `jira_issue_key` is `none` to decide whether to fetch JIRA issue details or not. This is a good practice as it avoids unnecessary API calls.
   - The use of environment variables like `$GITHUB_ENV` and `${{ secrets.CONFLUENCE_API_TOKEN }}` is a good practice to avoid hardcoding sensitive information in the code.

4. **Suggested Improvements or Refactoring:**
   - Consider refactoring the code to use a scripting language like Python. This would make the code more readable and maintainable, and could potentially improve performance.
   - Consider using a JIRA SDK or library instead of making raw API calls. This would provide a higher-level interface to interact with JIRA and could handle things like authentication and error handling for you.
   - Consider using a markdown library to create the `RELEASE_NOTES` string. This would be more reliable and easier to understand than using `sed` commands to process the string.
